packages/expo-gl-cpp/cpp/EXGLContext.cpp

```diff
@@ -0,0 +1,93 @@
+#include "EXGLContext.h"
+
+#include <JavaScriptCore/JSContextRef.h>
+
+#define STB_IMAGE_IMPLEMENTATION
+#include "stb_image.h"
+
+static std::unordered_map<UEXGLContextId, EXGLContext *> EXGLContextMap;
+static std::mutex EXGLContextMapMutex;
+static UEXGLContextId EXGLContextNextId = 1;
+
+std::atomic_uint EXGLContext::nextObjectId { 1 };
+
+EXGLContext *EXGLContext::ContextGet(UEXGLContextId exglCtxId) {
+  std::lock_guard<decltype(EXGLContextMapMutex)> lock(EXGLContextMapMutex);
+  auto iter = EXGLContextMap.find(exglCtxId);
+  if (iter != EXGLContextMap.end()) {
+    return iter->second;
+  }
+  return nullptr;
+}
+
+UEXGLContextId EXGLContext::ContextCreate(JSGlobalContextRef jsCtx) {
+  // Out of ids?
+  if (EXGLContextNextId >= std::numeric_limits<UEXGLContextId>::max()) {
+    EXGLSysLog("Ran out of EXGLContext ids!");
```

Comment:
Is it likely to happen? Should we account for that, eg. start from 0 and looking for an empty slot?

Reply:
it's highly unlikely to reach those values, I think that this check was added just handle all the cases. It would be probably hard to reach that even in an artificial test designed to break this.